test/test.js

```diff
@@ -174,13 +231,66 @@ describe('typeis.hasBody(req)', function () {
       assert.strictEqual(typeis.hasBody(req), true)
     })
   })
+
+  if (process.env.HTTP2_TEST) {
+    describe('http2 request', function () {
+      it('should not indicate body', function (done) {
+        createBodylessRequest('', function (req) {
+          assert.strictEqual(typeis.hasBody(req), false)
+          done()
+        })
+      })
+
+      it('should indicate body', function (done) {
+        createRequest('', function (req) {
+          assert.strictEqual(typeis.hasBody(req), true)
+          done()
+        })
+      })
+    })
+  }
 })
 
-function createRequest (type) {
-  return {
-    headers: {
-      'content-type': type || '',
-      'transfer-encoding': 'chunked'
+function createRequest (type, callback) {
+  if (process.env.HTTP2_TEST) {
+    var server = http2.createServer(function (req, res) {
+      callback(req)
+    })
+
+    server = server.listen(function () {
+      request.post('localhost:' + server.address().port + '/')
+        .send('hello')
+        .set('content-type', type || undefined)
+        .end()
+    })
+  } else {
+    var req = {
+      headers: {
+        'content-type': type || '',
+        'transfer-encoding': 'chunked'
+      }
+    }
+    callback(req)
+  }
+}
+
+function createBodylessRequest (type, callback) {
+  if (process.env.HTTP2_TEST) {
+    var server = http2.createServer(function (req, res) {
+      callback(req)
+    })
+
+    server = server.listen(function () {
```

Comment:
Don't forget to close the server, otherwise every test leaves a new server open, eventually going to starve the test memory over time.